mongo.y

```diff
@@ -94,6 +95,21 @@ span' p xs@(x:xs')
     | p x xs'      =  let (ys,zs) = span' p xs' in (x:ys,zs)
     | otherwise    =  ([],xs)
 
+floatRegex = "^-?[0-9]+\\.?[0-9]+"
```

Comment:
"^-?([0-9]+\\.[0-9]*|[0-9]*\\.[0-9]+)"
Pretty sure we had to come up with this for a pset in my compilers class :)